datasets/msnoise.py

```diff
@@ -237,44 +189,53 @@ def __reshape_audio(self, audio, row_len=128):
 
         return torch.transpose(torch.tensor(audio.reshape((-1, row_len))), 1, 0)
 
-    def __gen_datasets(self, exp_len=16384, row_len=128, overlap_ratio=0):
+    def __gen_datasets(self):
 
         with warnings.catch_warnings():
             warnings.simplefilter('error')
 
-            # PARAMETERS
-            overlap = int(np.ceil(row_len * overlap_ratio))
-            num_rows = int(np.ceil(exp_len / (row_len - overlap)))
-            data_len = int((num_rows*row_len - (num_rows-1)*overlap))
-            print(f'data_len: {data_len}')
-
-            # Cleaning the duplicate labels
+            labels = list(self.classes)
             train_list = sorted(os.listdir(self.noise_train_folder))
             test_list = sorted(os.listdir(self.noise_test_folder))
             labels_train = set(sorted({i.split('_')[0] for i in train_list if '_' in i}))
             labels_test = set(sorted({i.split('_')[0] for i in test_list if '_' in i}))
-            labels = labels_train | labels_test
-            labels_to_remove = set()
+
+            if self.d_type == 'train':
+                check_label = labels_train
+                audio_folder = [self.noise_train_folder]
+            elif self.d_type == 'test':
+                check_label = labels_test
+                audio_folder = [self.noise_test_folder]
+
             for label in labels:
-                other_labels = labels - {label}
-                for other_label_name in other_labels:
-                    if label in other_label_name:
-                        labels_to_remove.add(label)
-                        break
-            labels = labels - labels_to_remove
-            labels = sorted(labels)
-            print(f'Labels: {labels}')
-
-            # Folders
-            train_test_folders = [self.noise_train_folder, self.noise_test_folder]
+                if label not in check_label:
+                    print(f'Label {label} is not in the MSnoise {self.d_type} dataset.')
+                    labels.remove(label)
+
+            print(f'Labels for {self.d_type}: {labels}')
+
+            if self.desired_probs is None or len(self.desired_probs) != len(labels):
+                self.desired_probs = []
+                print('Each class will be selected using the same probability!')
+                label_count = len(labels)
+                for i in range(label_count):
+                    self.desired_probs.append(1/label_count)
+
+            elif np.sum(self.desired_probs) != 1:
+                print('Sum of the probabilities is not 1!')
```

Comment:
Also print a helpful note to the user, such as "Carrying out the proper normalization"